File: 프로그래머스/lv1/이하은/추억 점수.kt
import java.util.*

class Solution {
    fun solution(names: Array<String>, yearning: IntArray, photos: Array<Array<String>>): ArrayList<Int> {
        val answer = arrayListOf<Int>()
        
        val hash = HashMap<String, Int>()
        
        for((i, name) in names.withIndex()){
            hash[name] = yearning[i]
        }
        
        for(photo in photos){
            var score = 0
            for(name in photo){
                if(hash.containsKey(name)){
                    score += hash[name] ?: 0
                }
            }
            answer.add(score)
        }

        return answer
    }
}

File: 프로그래머스/lv1/김원래/추억 점수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

unordered_map<string, int> yearnings;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    
    for(int cnt = 0 ; cnt < name.size(); ++cnt)
    {
        yearnings.insert({name[cnt], yearning[cnt]});
    }
    
    for(int photoCnt = 0 ; photoCnt < photo.size() ; ++photoCnt)
    {
        int score = 0;
        for(string s : photo[photoCnt])
        {
            if(yearnings.find(s) != yearnings.end())
            {
                score += yearnings[s];
            }
        }
        
        answer.push_back(score);
    }
    
    return answer;
}

/*



*/